src/libsyntax_pos/lib.rs

```diff
@@ -668,6 +668,11 @@ impl MultiSpan {
         &self.primary_spans
     }
 
+    /// Returns wether any of the primary spans is displayable.
+    pub fn has_primary_spans(&self) -> bool {
+        self.primary_spans.iter().any(|sp| *sp != DUMMY_SP)
```

Comment:
Comparison with `DUMMY_SP` is almost never correct, it's preferable to use `span.is_dummy()`.

Reply:
Fixed